.github/workflows/publish-all-platforms.yml

```diff
@@ -168,3 +174,17 @@ jobs:
     secrets:
       BREEZ_MVN_USERNAME: ${{ secrets.BREEZ_MVN_USERNAME }}
       BREEZ_MVN_PASSWORD: ${{ secrets.BREEZ_MVN_PASSWORD }}
+
+  publish-react-native:
+    needs:
+      - setup
+      - publish-maven
```

Comment:
The dependency on `publish-maven` needs some additional work in the Setup step.

That would need in the setup stage for example:
```
maven: ${{ !!inputs.maven-package-version || !!inputs.react-native-package-version }}
```

Same goes for dependencies of `publish-maven`, so android and the specific language bindings you need.

One thing what will happen then is; the maven package will actually be published, because `publish`=true. But you probably don't want that to be published if you're only building react-native. Maybe in the setup stage we can make specific booleans for every platform. Like `publish-react-native`, `publish-kotlin-mpp`, `publish-maven`, etc. And pass those specific booleans in the correcponding `publish` parameters of each job. (if that makes sense)

Reply:
Makes sense, thanks for the explanation. 🙏 

Thinking of this again, the `publish-react-native` step doesn't strictly _need_ the `publish-maven` step to run beforehand. The `publish-react-native` step will success and the NPM package will be published but at runtime it will require the respective Android and iOS packages to be available. At the moment those are assumed to be released in lockstep: RN plugin x.y.z requires Android and iOS package x.y.z.

That's why I added the dependency at build time just to make it more explicit that publishing the RN package alone doesn't make sense. But I see now that this is not so easy to model here.

Would it make sense to add `maven: ${{ !!inputs.maven-package-version || !!inputs.react-native-package-version }}` like you suggested (also for all dependencies of that step) and then accept that all those will be build and published every time we publish react native?

What do you think?

Reply:
Ok reconsidering this again. I removed the build time dependencies, we cannot (yet) enforce it for ios anyway so it's not very useful to _just_ do it for maven I guess. I think for the time being the flexibility to publish RN alone is better. We just published 0.2.0 and probably want to publish RN 0.2.0. For now whoever publishes just needs to know that this runtime dependency exists. 

I can also write that down in somewhere in a releasing doc or so. Thoughts? (also cc @roeierez to keep you in the loop when you do the publishing)

Reply:
Interesting question. I guess it makes sense to not add the build dependency in the pipeline then if it's not strictly needed in order to publish a new version. It may lead to a small time of inconsistency if we publish the new react native version before publishing the new maven version, but I think that's ok?

Reply:
Agreed. I think that is ok. We just need to be careful when publishing, not to publish RN when there's not a corresponding Android/iOS package.